muspinsim/input/structure.py

```diff
@@ -377,3 +374,47 @@ def get_efg_tensor(self, atom_index: int) -> ArrayLike:
             efg_tensor {ArrayLike}: EFG tensor for the atom at the given index
         """
         return self._efg_tensors[atom_index - 1]
+
+    @property
+    def muon(self) -> List[str]:
+        """Returns the muon object from this structure"""
+        return self._cell_atoms[self._muon_index]
+
+    def move_atom(self, atom1: CellAtom, atom2: CellAtom, new_distance: float):
+        """Moves atom2 to be a specific distance away from atom1, while preserving
+        the direction between them
+
+        Arguments:
+            atom1 {CellAtom} -- Atom to compute the current distance from
+            atom2 {CellAtom} -- Atom that will be moved
+            new_distance {float} -- New distance that should be between the
+                                    atoms after moving
+
+        Raised:
+            NotImplementedError: If atom2 is the muon (All the other positions would
+                        need recalculating otherwise)
+        """
+        if atom2 == self.muon:
+            raise NotImplementedError("Moving the muon is not supported")
+
+        vector_between = atom2.position - atom1.position
+        distance = np.linalg.norm(vector_between)
+
+        # old_pos + new_dist * direction
+        new_pos = atom1.position + ((new_distance / distance) * vector_between)
+
+        # Log what is happening so can keep track
+        logging.info(
+            "Moving %s from %s to %s, changing the distance from %s to %s",
+            atom2.symbol,
+            atom2.position,
+            new_pos,
+            distance,
+            new_distance,
+        )
+
+        atom2.position = new_pos
+
+        # Update the distance to the muon if necessary
+        if atom1 == self.muon:
+            atom2.distance_from_muon = new_distance
```

Comment:
Perhaps I'm not fully understanding where/how in the process we would call this, but wouldn't we need to recalculate the distance to the muon regardless of whether atom 1 is the muon?  From looking at the script in dev notes, we're continuing to use the `distance_from_muon` after initially setting the NN `F` to 1.21, so makes sense that we need this property to be accurate for the later stages, and since we're always moving relative to the muon, it doesn't matter there.

But generally speaking: say I have a `mu` at `0 0 0`, atom `A` atom `1 0 0` and atom B at `2 0 0`. I rescale `A` and `B` to have a distance of 2 between them, so now we have `0 0 0`, `1 0 0` and `3 0 0` respectively. In this case `atom1` was not the muon but the distance from the muon to `B` will have increased from 2 to 3?

As I said I might be misunderstanding the situation and I don't think it will matter for the ScF3 case - but would rather the distance is always accurate in case a future use case needs it (and isn't moving relative to the muon).

Reply:
Yes you're right, just addressed this.